aderyn_core/src/detect/low/state_variables_not_logged_in_event.rs

```diff
@@ -0,0 +1,229 @@
+use std::{
+    collections::{BTreeMap, HashMap},
+    error::Error,
+};
+
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::{
+    ast::{ASTNode, Expression, Mutability, NodeID, VariableDeclaration},
+    context::{
+        browser::{ExtractEmitStatements, GetImmediateChildren},
+        workspace_context::WorkspaceContext,
+    },
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use crate::{capture, context::browser::ExtractAssignments};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct StateVariableNotLoggedInEventDetector {
+    // All the state variables, set at the beginning of the detect Function
+    mutable_state_variables: HashMap<i64, VariableDeclaration>,
```

Comment:
I don't think `mutable_state_variables` needs to be a struct variable. It seems more appropriate to make it a local variable in the function `detect()`. 

```rust
    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
        ...
        let mutable_state_variables: HashMap<i64, VariableDeclaration> = context
            .variable_declarations()
            ...
```